### ISSUE TITLE
Accessing PDF FIle

### ISSUE THREAD

**Issue by madeuserforaquestion** [fdemmer/django-weasyprint#33]
Hello, 
I currently have a PDF generator where a user can generate a unique PDF based on certain data they input. I am trying to save the PDF to the server file system but I'm not too sure on how to access the file and where would be the best place access it method wise. I was thinking about not using an attachment but each PDF is unique and the values not stored anywhere. Here is my pseudo code:
```
class MyDownloadAttachmentView(WeasyTemplateResponseMixin, DetailView):
    pdf_attachment = True

    # what I am trying to do
    def render_to_reponse(self):
        pdf = self.file  # basically the unique PDF that is being rendered from the view
        artifact.objects.create(    
           name='pdf_artifact',
           file=pdf
        )  # create an artifact with the PDF
```
Basically I am looking for a way to access the PDF file object and save it to my local file system. Thanks in advance

**Comment by madeuserforaquestion** [comment]
I solved the issue. Final code:
```
class MyDownloadAttachmentView(WeasyTemplateResponseMixin, DetailView):
    pdf_attachment = True
    def render_to_response(self, context, **response_kwargs):
        response_kwargs.update({
            'attachment': self.pdf_attachment,
            'filename': self.get_pdf_filename(),
            'stylesheets': self.get_pdf_stylesheets(),
        })
        res = super(MyDownloadAttachmentView, self).render_to_response(
            context, **response_kwargs
        )
        doc = res.get_document().write_pdf()
        f = ContentFile(doc)
        a = models.Artifact(
            objt_id=self.object,
            name=self.get_pdf_filename(),
            notes='Report auto uploaded on creation'
        )
        a.file.save(f'{self.get_pdf_filename()}.pdf', f)
        a.save()
        return res
```
Need to clean up the code and but it works. Self closing the issue.

**Comment by ashishnitinpatil** [comment]
Just in case someone stumbles on this to access the PDF via another django view, I took inspiration from the above solution and got the PDF content with below code -

```python
from django.http import HttpRequest
from django.core.files.base import ContentFile


def pdf_magic(request: HttpRequest, my_model_object: MyModel):
    new_request = HttpRequest()
    new_request.method = "GET"
    new_request.user = request.user
    new_request.META = request.META
    response = MyModelPDFView.as_view()(
        new_request, slug=my_model_object.slug  # replace slug with anything else that your view may require
    )
    return ContentFile(response.rendered_content)
```

You can then call the above function in your view and save the file as follows -
```python
...
my_model_object.file.save("filename.pdf", pdf_magic(request, my_model_object), save=False)
my_model_object.save()
return response
```

(more explanation for better SEO)
I'm calling another django view from a view and then using the rendered response's file contents.
